test/jdk/javax/swing/JFileChooser/4400728/JFileChooserDefaultDirectoryTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2004, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.File;
+import javax.swing.JFileChooser;
+
+/*
+ * @test
+ * @bug 4400728
+ * @summary This testcase tests CCC4400728 request, checks whether JFileChooser
+ *          constructor set correct default directory or not, it is typically
+ *          the "My Documents" folder on Windows, and the user's home directory
+ *          on Unix.
+ * @run main JFileChooserDefaultDirectoryTest
+ */
+public class JFileChooserDefaultDirectoryTest {
+
+    public static void main(String[] args) {
+        JFileChooser jFileChooser = new JFileChooser();
+        final String actualDefaultDirectory =
+                jFileChooser.getFileSystemView().getDefaultDirectory()
+                            .getName();
+        final boolean isWindows = System.getProperty("os.name").toLowerCase()
+                                        .startsWith("windows");
+        if (isWindows) {
+            if (actualDefaultDirectory.endsWith("Documents")) {
```

Comment:
Based on this PR https://github.com/openjdk/jdk/pull/7983 can we check jus the "Documents" string?

Reply:
As Windows 8.x is still supported, isn't it better to do it like this, for safer side?

Reply:
Ok, not a big deal.

Reply:
"My Documents" changed to "Documents" in Windows Vista. All the supported Windows versions use "Documents" as the folder name.

Reply:
In fact, users can move the Documents folder to an arbitrary location, yet I don't expect many do.

I wanted to suggest making a stronger assertion on the path: `System.getProperty("user.home") + File.separator + "Documents"`. Yet it may fail if the default folder is relocated.

Reply:
I have changed it to actualDefaultDirectory.equals("Documents"), but in Windows case, we are just checking the default directory name(last name), not the full path, so does this hold good?

Reply:
I understand that you verify only the last part of the path. It's the safest approach.

The default full path would be `System.getProperty("user.home") + File.separator + "Documents"`. Yet it may not be the actual path for Documents folder because users can relocate it. For example with OneDrive installed, the default folder could reside inside OneDrive, the absolute path will be different in this case.

Reply:
I think `actualDefaultDirectory.equals("Documents")` always returns false.

My sentence above was rather a confirmation that you're doing the right thing and comparing the full path as you do for other platforms wouldn't work.

Reply:
Ok, thank you